src/platform/plugins/shared/chart_expressions/expression_partition_vis/public/utils/layers/get_color.test.ts

```diff
@@ -19,371 +19,390 @@ import { ChartTypes } from '../../../common/types';
 import { getDistinctSeries } from '..';
 import { getColorCategories } from '@kbn/chart-expressions-common';
 
-describe('#byDataColorPaletteMap', () => {
-  let paletteDefinition: PaletteDefinition;
-  let palette: PaletteOutput;
-  let colorIndexMap: Map<string, number>;
-  const visData = createMockVisData();
-  const categories = (chartType?: ChartTypes) =>
-    chartType === ChartTypes.MOSAIC && visData.columns.length === 2
-      ? getColorCategories(visData.rows, visData.columns[1]?.id)
-      : getColorCategories(visData.rows, visData.columns[0]?.id);
+describe('get color', () => {
+  describe('#byDataColorPaletteMap', () => {
+    let paletteDefinition: PaletteDefinition;
+    let palette: PaletteOutput;
+    let colorIndexMap: Map<string, number>;
+    const visData = createMockVisData();
+    const categories = getColorCategories(visData.rows, visData.columns[1]?.id);
 
-  beforeEach(() => {
-    paletteDefinition = chartPluginMock.createPaletteRegistry().get('default');
-    palette = { type: 'palette' } as PaletteOutput;
-    colorIndexMap = new Map(categories().map((d, i) => [d[0], i]));
-  });
+    beforeEach(() => {
+      paletteDefinition = chartPluginMock.createPaletteRegistry().get('default');
+      palette = { type: 'palette' } as PaletteOutput;
+      colorIndexMap = new Map(categories.map((c, i) => [String(c), i]));
+    });
 
-  it('should create byDataColorPaletteMap', () => {
-    expect(byDataColorPaletteMap(paletteDefinition, palette, colorIndexMap)).toMatchInlineSnapshot(`
-      Object {
-        "getColor": [Function],
-      }
-    `);
-  });
+    it('should create byDataColorPaletteMap', () => {
+      const colorPaletteMap = byDataColorPaletteMap(paletteDefinition, palette, colorIndexMap);
+      expect(colorPaletteMap.getColor).toEqual(expect.any(Function));
+    });
 
-  it('should get color', () => {
-    const colorPaletteMap = byDataColorPaletteMap(paletteDefinition, palette, colorIndexMap);
+    it('should get color', () => {
+      const colorPaletteMap = byDataColorPaletteMap(paletteDefinition, palette, colorIndexMap);
 
-    expect(colorPaletteMap.getColor('Logstash Airways')).toBe('black');
-  });
+      expect(colorPaletteMap.getColor('Logstash Airways')).toBe('black');
+    });
 
-  it('should return undefined in case if values not in datatable', () => {
```

Comment:
I don't see this test as being necessary as returning a `colorIndex` of `-1` is the same as returning `undefined`.